python/pyspark/ml/param/__init__.py

```diff
@@ -65,6 +72,106 @@ def __eq__(self, other):
             return False
 
 
+class TypeConverters(object):
+    """
+    .. note:: DeveloperApi
+
+    Factory methods for common type conversion functions for `Param.typeConverter`.
+
+    .. versionadded:: 2.0.0
+    """
+
+    @staticmethod
+    def _is_numeric(value):
+        vtype = type(value)
+        return vtype == int or vtype == float or vtype == np.float64 \
+            or vtype == np.int64 or vtype.__name__ == 'long'
+
+    @staticmethod
+    def _can_convert_to_list(value):
+        vtype = type(value)
+        return vtype == list or vtype == np.ndarray or isinstance(value, Vector)
+
+    @staticmethod
+    def identity(value):
+        """
+        Dummy converter that just returns value.
+        """
+        return value
+
+    @staticmethod
+    def convertToList(value):
```

Comment:
Naming: How about everything be called ```toX``` for type ```X```?  It's clear it's for converting b/c of the class name.